service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java

```diff
@@ -139,37 +164,60 @@ public void configPush(CommandRequest request) {
             cleanup(executionLogFilesToClean);
         }
+    }
 
+    // Attaches the error log to the first terraform command in the request to be sent back to EP although the error
+    // occurred during the deletion of the terraform state file and not during the terraform command execution
+    private void attachErrorLogToFirstTerraformCommand(RuntimeException ex, CommandRequest request){
+        for (CommandBundle bundle : request.getCommandBundles()) {
+            for (Command command : bundle.getCommands()) {
+                if (command.getCommandType() == terraform) {
+                    CommandResult errorResult = terraformLoggingService.buildTfStateFileDeletionFailureResult(ex);
+                    command.setResult(errorResult);
+                    return;
+                }
+            }
+        }
+        log.error("No terraform command found to attach error log to");
     }
 
-    private Path executeCommand(CommandRequest request,
-                                Command command,
-                                Map<String, String> envVars) {
+    private Path executeTerraformCommand(CommandRequest request, Command command, Map<String, String> envVars) {
         Path executionLog = null;
         try {
-            switch (command.getCommandType()) {
-                case terraform:
-                    executionLog = terraformManager.execute(request, command, envVars);
-                    break;
-                default:
-                    command.setResult(CommandResult.builder()
-                            .status(JobStatus.error)
-                            .logs(List.of(
-                                    Map.of("message", "unknown command type " + command.getCommandType(),
-                                            "errorType", "UnknownCommandType",
-                                            "level", LOG_LEVEL_ERROR,
-                                            "timestamp", OffsetDateTime.now())))
-                            .build());
-                    break;
-            }
+            Validate.isTrue(command.getCommandType().equals(terraform), "Command type must be terraform");
+            return terraformManager.execute(request, command, envVars);
         } catch (Exception e) {
-            log.error("Error executing command", e);
+            log.error(ERROR_EXECUTING_COMMAND, e);
             setCommandError(command, e);
-        }
 
+        }
         return executionLog;
     }
 
+    private void executeSempCommand(Command command, SolaceHttpSemp solaceClient) {
+        try {
+            Validate.isTrue(command.getCommandType().equals(semp), "Command type must be semp");
+            // only delete operation is supported for now and only delete operations are sent from EP
+            Validate.isTrue(command.getCommand().equals(SEMP_DELETE_OPERATION), "Command operation must be delete");
+            sempDeleteCommandManager.execute(command, new SempApiProviderImpl(solaceClient));
```

Comment:
Instead of creating a new `SempApiProviderImpl` for each command, we could just change the method signature here as 
`private void executeSempCommand(Command command, SempApiProviderImplsempApiProvider) {`
create a single SempApiProviderImpl on line 138